app/controllers/admin/links_controller.rb

```diff
@@ -1,45 +1,44 @@
 # frozen_string_literal: true
 
 class Admin::LinksController < Admin::BaseController
-  before_action :fetch_product_by_general_permalink, except: %i[legacy_purchases
-                                                                flag_seller_for_tos_violation
-                                                                views_count sales_stats
-                                                                join_discord
-                                                                join_discord_redirect]
-  before_action :fetch_product, only: %i[views_count
-                                         sales_stats
-                                         join_discord
-                                         join_discord_redirect]
+  before_action :fetch_product!, except: :show
```

Comment:
Not a fan of using `except` in general, but this is the one case where it makes sense.